snapshot/db.go

```diff
@@ -4,43 +4,28 @@ import (
 	"github.com/scootdev/scoot/snapshot/git/repo"
 )
 
-// ID identifies a Value in DB. (Cf. doc.go for explanation of Value) Opaque to the client.
+// ID identifies a Snapshot in DB. (Cf. doc.go for explanation of Snapshot) Opaque to the client.
 type ID string
 
-// DB is the Scoot Database, allowing creation, distribution, and export of Values.
-type DB interface {
+// Creator allows creating new Snapshots.
+type Creator interface {
 	// Ingest
 
 	// IngestDir ingests a directory directly.
-	// The created Value is a Snapshot.
+	// The created Snapshot is an FSSnapshot.
 	IngestDir(dir string) (ID, error)
 
 	// IngestGitCommit ingests the commit identified by commitish from ingestRepo
 	// commitish may be any string that identifies a commit
-	// The created Value is a SnapshotWithHistory.
+	// The created Snapshot is a GitCommitSnapshot.
 	IngestGitCommit(ingestRepo *repo.Repository, commitish string) (ID, error)
+}
 
-	// Operations
-
-	// UnwrapSnapshotHistory unwraps a SnapshotWithHistory and returns a Snapshot ID.
-	// Errors if id does not identify a SnapshotWithHistory.
-	UnwrapSnapshotHistory(id ID) (ID, error)
-
-	// Distribute
-
-	// Upload makes sure the Value id is uploaded, returning an ID that can be used
-	// anywhere or an error
-	Upload(id ID) (ID, error)
-
-	// Download makes sure the Value id is downloaded, returning an ID that can be used
-	// on this computer or an error
-	Download(id ID) (ID, error)
-
-	// Export
-
-	// Checkout puts the Value identified by id in the local filesystem, returning
+// Reader allows reading data from existing Snapshots
+type Reader interface {
```

Comment:
on the fence as to whether this should be called Checkouter vs Reader.... if all methods use the word 'checkout', then I would lean toward Checkouter.